src/Google.Pubsub.V1/SimplePubsub.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.Api.Gax;
+using Grpc.Core;
+using System;
+using System.Threading.Tasks;
+
+using static Google.Pubsub.V1.PublisherClient;
+using static Google.Pubsub.V1.SubscriberClient;
+
+// TODOs for this, SimpleSubscription and SimpleTopic:
+// - Naming (the "simple" prefix isn't great)
+// - Consider cancellation for synchronous methods
+// - Documentation
+// - More tests
+// - Validate that using the IDs instead of resource names is useful
+// - Convert demo code to use it
+
+namespace Google.Pubsub.V1
+{
+    /// <summary>
+    /// Simplified wrapper around the Google Pubsub API.
+    /// </summary>
+    public class SimplePubsub
+    {
+        /// <summary>
+        /// The ID of the project containing all the topics and subscriptions this instance works with.
+        /// </summary>
+        public string ProjectId { get; }
+
+        public SubscriberClient Subscriber { get; }
+        public PublisherClient Publisher { get; }
+
+        public SimplePubsub(string projectId, SubscriberClient subscriber, PublisherClient publisher)
+        {
+            ProjectId = projectId;
+            Subscriber = subscriber;
+            Publisher = publisher;
+        }
+
+        public static SimplePubsub CreateFromDefaultCredentials(string projectId)
+        {
+            GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
+            return new SimplePubsub(projectId,
+                SubscriberClient.CreateFromDefaultCredentials(),
+                PublisherClient.CreateFromDefaultCredentials());
+        }
+
+        public static async Task<SimplePubsub> CreateFromDefaultCredentialsAsync(string projectId)
+        {
+            GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
+            return new SimplePubsub(projectId,
+                await SubscriberClient.CreateFromDefaultCredentialsAsync(),
+                await PublisherClient.CreateFromDefaultCredentialsAsync());
```

Comment:
In general, yes. In this case, no - because once the default credentials have been fetched the first time, the second call will be trivial and effectively complete synchronously.